modal/image.py

```diff
@@ -1065,11 +1065,13 @@ def from_gcp_artifact_registry(
         ```python
         modal.Image.from_gcp_artifact_registry(
             "us-east1-docker.pkg.dev/my-project-1234/my-repo/my-image:my-version",
-            secrets=[modal.Secret.from_name("my-gcp-secret")],
+            secret=modal.Secret.from_name("my-gcp-secret"),
             add_python="3.11",
         )
         ```
         """
+        if "secrets" in kwargs:
+            raise ValueError("Passing a list of 'secrets' is not supported; use the singular 'secret' argument.")
```

Comment:
Nit: normally a `TypeError` is raised when you pass a parameter that a function does not accept.